### ISSUE TITLE
Make automatic inclusion Mixin optional

### ISSUE THREAD

**Issue by kikonen-fiksu** [TwP/logging-rails#18]
I like a lot logging and also logging-rails due to that. However, I don't particularly like automagical inclusion of "logger" everywhere via  ::Logging::Rails::Mixin done in https://github.com/TwP/logging-rails/blob/master/lib/logging/rails/railtie.rb.

While this is not problem with small project, with larger projects is starts to blur things a bit and creates hidden extra dependency.

So I would see beneficial to have option to turn off auto mixin, and have module, which would explicitly include "logger" method in the context.

For example,
````ruby
class Foobar
  include Logging::Log
end
```

**Comment by sshaw** [comment]
I agree, out the box `logging-rails` turns on the firehose. It should only add the logger to things that Rails does by default. This does not include [`Rails.cache`](https://github.com/TwP/logging-rails/blob/logging-rails-0.5.0/lib/logging/rails/railtie.rb#L43) and [`ActiveSupport::Dependencies.logger`](https://github.com/TwP/logging-rails/blob/logging-rails-0.5.0/lib/logging/rails/railtie.rb#L39).

It also (somehow) adds the logger to seemingly random classes at startup. Here's how logging configures my app:

```
root  ............................................  *debug      -T
- <Appenders::Stdout:0x3fcb793bf02c name="stdout">
- <Appenders::RollingFile:0x3fcb793beba4 name="file">
  ActionController::Base  ........................   debug  +A  -T
  ActionMailer::Base  ............................   debug  +A  -T
  ActiveRecord::Base  ............................   debug  +A  -T
  ActiveRecord::Relation  ........................   debug  +A  -T
  ActiveSupport::Cache::FileStore  ...............   debug  +A  -T
  ActiveSupport::Dependencies  ...................   debug  +A  -T
  ActiveSupport::OrderedOptions  .................   debug  +A  -T
  Category  ......................................   debug  +A  -T
  Country  .......................................   debug  +A  -T
  Logging  .......................................    *off  -A  -T
  Rails  .........................................   debug  +A  -T
```

Note `ActiveSupport::OrderedOptions`,  `Country`, and `Category`. The latter two are 2 of many models, not sure why they're shown.

`ActiveSupport::OrderedOptions` is random, and note that the messages it products are confusing:

```
[2015-05-10 15:18:53] INFO  ActiveSupport::OrderedOptions : Served asset /ember-1.0.0.min.js - 304 Not Modified (3ms)
```

That comes from Sprockets, not `OrderedOptions`.

Of course logging for these classes can be turned off, but I why even turn them on?
